docs/profiling.md

```diff
@@ -0,0 +1,31 @@
+# vHive profiling tool
+
+## Methodology
+
+The benchmark loads a certain amount of requests per second (RPS) to vHive for a time duration.
+
+The load procedure is split into different steps from 5% to 100% of input RPS. If the tail latency at a load step violates the constraint which is 5x image service time, the tool will return the profiling result from one step before the tail latency violation.
+
+A load step divides into three parts: warm-up, profiling and cool-down. During the profile period, the benchmark records the average execution time of invocations and how many invocations return successfully. It also invokes [pmu-tools](https://github.com/andikleen/pmu-tools) to monitor user-defined nodes followed by [TopDown method](https://ieeexplore.ieee.org/document/6844459). After the measurement finishes, successful RPS per core, average execution time and the average counters of nodes will save in the `profile.csv`.
+
+## Profile fixed number of VMs and RPS
+
+Profile a single VM with `helloworld` image and 20 RPS at TopDown level 1 :
+```
+sudo env "PATH=$PATH" go test -v -timeout 99999s -run TestProfileSingleConfiguration -args -funcNames helloworld -vm 1 -rps 20 -l 1
```

Comment:
all these runtime arguments need to be explained in the previous section.

Reply:
Done